pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go

```diff
@@ -203,6 +214,51 @@ type ReplicaResourceRequirement struct {
 	LuaScript string `json:"luaScript"`
 }
 
+// ComponentResourceRequirement holds the Lua script for extracting the desired replica count
```

Comment:
Here's a small suggestion: consider removing Lua. Firstly, Lua might just be one of the scripts we design in the future. Secondly, it would help maintain consistent commenting formats with other interfaces.